modules/scripts/startup-script/files/running-script-warning.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/sh
+
+FINISH_LINE="startup-script exit status"
+ACTIVE=^activ
+WARNING=$(
+	cat <<EOF
+** WARNING **: The VM startup scripts for this have not been completed and
+system services may not be configured yet.
+
+Attempting to make changes to the system may lead to undefined behavior.
+It is advised that you wait for the startup scripts to complete.
+An alert will be written to all logged in users confirming that the startup
+scripts have completed.
+
+Another way to check the status of the startup scripts is to run:
+$ systemctl status google-startup-scripts.service
+
+If the "Active" status is no longer "active", the scripts have completed.
+EOF
+)
+ERROR_NOTICE=$(
+	cat <<EOF
+** ERROR **: The VM startup scripts have finished running, but produced an error. 
+
+Please see GCP Cloud Console "Logs Explorer" tool or journalctl to determine
+the cause of the startup script failure.
+
+The following command can also help display relevant information.
+
+Local shell:
+$ sudo journalctl -u google-startup-scripts.service
+EOF
+)
+
+# Check if end of startup scripts has been detected
+IS_ACTIVE=$(systemctl is-active google-startup-scripts.service | grep "${ACTIVE}")
+END_FOUND=$(journalctl -b 0 -u google-startup-scripts.service | grep "${FINISH_LINE}")
```

Comment:
Nothing stops a user from starting google-startup-scripts multiple times after a boot. In fact, this is one way we might choose to help the user propagate Ansible updates. Is there a way to limit this command to the most recent activation of the google-startup-scripts.service unit.

(I did not see one in a quick scan)

Reply:
Well, I'm worried about defining a behavior. Do we target the boot-time activation of the service? Do we target all? Are we reporting success when the boot-time activation succeeds but subsequent activations fail?

And so-on. We should define the behavior and then confirm the code implements it.